### ISSUE TITLE
Consul fails to start with Unix syslog delivery error

### ISSUE THREAD

**Issue by ChrisAubuchon** [hashicorp/consul#1610]
Seeing this error occasionally starting consul:

```
consul: ==> Syslog setup failed: Unix syslog delivery error
```

This is with consul 0.6.1.

**Comment by shanielh** [comment]
Happens to me too with consul 0.6, I work in containerized environment so sometimes the consul container starts before the syslog container and in those cases the consul fails to start :

```txt
Attempting join: 10.0.0.250
Error connecting to Consul agent: dial tcp 127.0.0.1:8400: getsockopt: connection refused
==> Syslog setup failed: Unix syslog delivery error
Attempting join: 10.0.0.250
Error connecting to Consul agent: dial tcp 127.0.0.1:8400: getsockopt: connection refused
```

last two lines repeats every few seconds until I restart the consul container and then it works: 

```
 * Starting enhanced syslogd rsyslogd
   ...done.
10.0.0.2
Attempting join: 10.0.0.250
Error connecting to Consul agent: dial tcp 127.0.0.1:8400: getsockopt: connection refused
==> Starting Consul agent...
==> Starting Consul agent RPC...
==> Consul agent running!
         Node name: 'ip-10-0-0-159.ec2.internal'
        Datacenter: 'dc1'
            Server: false (bootstrap: false)
       Client Addr: 0.0.0.0 (HTTP: 8500, HTTPS: -1, DNS: 53, RPC: 8400)
      Cluster Addr: 10.0.0.159 (LAN: 8301, WAN: 8302)
    Gossip encrypt: false, RPC-TLS: false, TLS-Incoming: false
             Atlas: <disabled>
```

**Comment by micahlmartin** [comment]
:+1: Same here

**Comment by slackpad** [comment]
Need to think about the right solution for this one - maybe just a brief waiting period with a backoff?

**Comment by looprock** [comment]
I'm starting to see this too on several systems. Are there any work-arounds?

**Comment by looprock** [comment]
Interestingly, even when I roll back to 0.5.2, I still have the same issue. I even tried force-leaving the node and purging the data directory.

**Comment by evan2645** [comment]
Hit this problem with 0.6.3 today. Consul stop/start recovered it

**Comment by lin-zhao** [comment]
Hitting this with 0.6.3. rsyslog is actually running on port 514.

**Comment by evan2645** [comment]
present in 0.6.4 also. got to a spot where this was occurring every time a host was rebooted. worked around with aggressive respawn in upstart

**Comment by gsy** [comment]
Same problem with 0.6.3.  restart rsyslog and then restart consul make it work

**Comment by grindthemall** [comment]
Just hit this one today - running Centos 7.2, Rsyslog 8.10 and Consul .7.

Had to restart to enable an option, and it wouldn't restart with the message:

==> Syslog setup failed: Unix syslog delivery error

My systemd file already tries to auto-restart it quickly, but it does not matter.  I have one node now that I cannot get to run consul.

Suggestions?

**Comment by falzm** [comment]
Can this bug please get some attention? It's *really* annoying.

**Comment by kyhavlov** [comment]
Fixed with #2455
